### PR TITLE
isHappyNumber complete

### LINKED ISSUE #596: Fix isHappyNumber function
There is currently a method called `isHappyNumber` that lives in `/lib/isHappyNumber.js`.
It is incomplete and needs to be fixed!

According to [wikipedia](https://en.wikipedia.org/wiki/Happy_number):

> A **happy number** is a number defined by the following process: Starting with any positive integer, replace the number by the sum of the squares of its digits in base-ten, and repeat the process until the number either equals 1 (where it will stay), or it loops endlessly in a cycle that does not include 1. Those numbers for which this process ends in 1 are happy numbers, while those that do not end in 1 are unhappy numbers (or sad numbers).

### Acceptance Criteria:
1. Running `yarn test isHappyNumber` results in tests passing.
2. You have written a skeleton method for someone else to work on.
3. You have written tests surrounding your skeleton method.
4. Running `yarn lint` does not print any errors to the console!
5. Optional: write one or two more tests surrounding the method you are solving to account for potential edge cases.

### Please include the skeleton of a new method + an accompanying test for someone else to work on at the time of creating a pull request! A pull request will most likely be denied if it does not contain a skeleton method for someone else to work on! For more information, please read the [Contributing Guide](https://github.com/BlakeGuilloud/ganon/blob/master/CONTRIBUTING.md).

Thank you so much for your contribution!

### PR DESCRIPTION
**Pull request by Soben**
Closes #596

- [x] Running `yarn lint` does not trigger any linter errors
- [x] The test of the method you have fixed is passing
- [x] You have written a new skeleton method for someone else to work on!
- [x] You have written tests to accompany your skeleton method

### CODE REVIEW

**Comment by Soben** [comment]
Skeleton "isEmailAddress" with test, completed.

File: test/isEmailAddress.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  test("Is Email Address (boogie+label@woogie.com) ?", () => {
+    expect(isEmailAddress("boogie+label@woogie.com")).toBe(true);
+  });
+
+  test("Is Invalid Email Address (boogiewoogie.com) ?", () => {
+    expect(isEmailAddress("boogiewoogie.com")).toBe(false);
+  });
+
+  test("Is Invalid Email Address (boogie @ woogie. noogie) ?", () => {
+    expect(isEmailAddress("boogie @ woogie. fo' noogie")).toBe(true);
```
Comment (BlakeGuilloud) [inline test/isEmailAddress.test.js]: is this supposed to be `toBe(false)` or is the test description incorrect?

Comment (Soben) [reply test/isEmailAddress.test.js]: SHIT. yea. you're right. fixing now. it'll be up in a moment.

**Comment by davcs86** [comment]
I want the isEmailAddress when gets available, please.
